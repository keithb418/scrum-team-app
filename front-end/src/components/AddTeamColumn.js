import React from "react";
import { Button } from "react-bootstrap";
import FontAwesome from "react-fontawesome";

const AddTeamColumn = () => {
  return (
    <Button className="add-team-button">
      <span className="content">
        <FontAwesome
          name="plus-circle"
        />
        Add Team Column
      </span>
    </Button>
  );
};

export default AddTeamColumn;
