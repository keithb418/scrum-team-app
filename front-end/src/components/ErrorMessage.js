import React from "react";
import { Alert } from "react-bootstrap";

const ErrorMessage = ({ message }) =>
  message
    ? <div><Alert bsStyle="warning"><strong>Error:</strong> {message}</Alert></div>
    : null;

export default ErrorMessage;