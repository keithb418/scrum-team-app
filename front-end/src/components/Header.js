import React from "react";
import { Row, Col } from "react-bootstrap";

const Header = () => {
  return (
    <header>
      <Row>
        <Col xs={12}>
          <div className="page-header">
            <h1>Scrum Team</h1>
          </div>
        </Col>
      </Row>
    </header>
  );
};

export default Header;
