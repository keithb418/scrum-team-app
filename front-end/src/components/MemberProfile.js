import React, { Component } from "react";
import { connect } from "react-redux";
import { Button, Row, Col, Panel } from "react-bootstrap";
import { fetchProfileData } from "../actions/shared";
import FontAwesome from "react-fontawesome";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import { capitalizeFirstLetter } from "../utils/stringHelpers";
import Loading from "./Loading";
import Profile from "./Profile";
import Skills from "./Skills";

export class MemberProfile extends Component {
  constructor () {
    super();
  }
 
  componentDidMount () {
    this.props.fetchProfileData(this.props.match.params.id);
  }
 
  render () {
    const { teamMember, team } = this.props;
    return !teamMember ?
        <Loading /> :
        <Row>
          <Col className="profile-section" md={6} mdOffset={3} xs={12}>
            <Panel.Heading>
              <h3 className="panel-title">{team.name}</h3>
            </Panel.Heading>
            <hr />
            <div className="member-profile-subheading">
              <FontAwesome name="user-circle" className="member-img" />
              <p>{teamMember.name}</p>
            </div>
            <hr />
            <Row>
              <Profile 
                member={teamMember}/>
              <Skills 
                member={teamMember}/>
            </Row>
            <hr />
            <Link to={`/member/${teamMember._id}/edit`}>
              <Button bsStyle="primary">Edit</Button>
            </Link>
            <Link to="/">
              <Button>Back to Dashboard</Button>
            </Link>
          </Col>
        </Row>;
  }
};

MemberProfile.defaultProps = {
  team: {
    name: ""
  },
  isFetching: true
};

MemberProfile.propTypes = {
  teamMember: PropTypes.object,
  team: PropTypes.object
};

const mapStateToProps = ({ teamMembers: { teamMember }, teams: { teams } }) => {
  if(teamMember) {
    const team = teams.find(team => {
      return team._id === teamMember.team;
    });
    return {
      team,
      teamMember
    };
  } else {
    return {
      teamMember
    };
  }  
};

const mapDispatchToProps = (dispatch) => ({
  fetchProfileData: (id) => dispatch(fetchProfileData(id))
});


export default connect(mapStateToProps, mapDispatchToProps)(MemberProfile);