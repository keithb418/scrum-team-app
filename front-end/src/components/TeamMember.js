import React from 'react';
import { Button, Row, Col  } from 'react-bootstrap';
import FontAwesome from 'react-fontawesome';

const TeamMember = ({ _id: id, name, teamLead, role }) => {
    let teamLeadText = teamLead ? <p>Team Lead</p> : '';

    return (
        <Button bsStyle={teamLead ? 'info' : 'default'} className="team-member">
            <Row>
                <Col xs={4} className="user-img">
                    <FontAwesome name="user-circle" />
                </Col>
                <Col xs={8} className="details">
                    <p>{name}</p>
                    {teamLeadText}
                    <p>{role}</p>
                    <FontAwesome name="edit" />
                </Col>
            </Row>
        </Button>
    );
};

export default TeamMember;