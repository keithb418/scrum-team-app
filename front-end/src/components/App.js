import React from "react";
import TeamColumnsContainer from "../containers/TeamColumnsContainer";
import AddTeamMember from "./forms/AddTeamMember";
import Header from "./Header";

const App = () => (
  <div>
    <Header />
    <TeamColumnsContainer />
    <AddTeamMember />
  </div>
);

export default App;
