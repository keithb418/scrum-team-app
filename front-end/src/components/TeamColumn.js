import React from "react";
import { connect } from "react-redux";
import TeamMember from "./TeamMember";
import TeamHeader from "./TeamHeader";

let TeamColumn = ({ id = "", teamName = "" , project = "", teamMembers = [], dispatch }) => {

  teamMembers.sort((a, b) => {
    if (b.teamLead) {
      return 1;
    }

    if (a.teamLead) {
      return -1;
    }

    return 0;
  });

  const teamMemberComponents = teamMembers.map((teamMember) =>
    <TeamMember 
      key={teamMember._id}
      id={teamMember._id} 
      name={teamMember.name} 
      teamLead={teamMember.teamLead} 
      role={teamMember.role} />
  );

  let allowDrop = (e) => {
    e.preventDefault();
  };

  let drop = (e) => {
    e.preventDefault();

    dispatch({
      type: "CHANGE_TEAM",
      teamMemberId: e.dataTransfer.getData("tmId"),
      team: id
    });
  };

  return (
    <div className="panel panel-default team-column" onDragOver={allowDrop} onDrop={drop}>
      <TeamHeader 
        id={id}
        teamName={teamName} 
        project={project} />
      <div className="panel-body team-body" onDrop={drop} onDragOver={allowDrop} >
        {teamMemberComponents}
      </div>
    </div>
  );
};

TeamColumn = connect()(TeamColumn);

export default TeamColumn;