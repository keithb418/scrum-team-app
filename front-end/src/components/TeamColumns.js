import React from "react";
import TeamColumn from "./TeamColumn";
import AddTeamColumn from "./AddTeamColumn";

const TeamColumns = ({ teams }) => {
  
  let teamColumns = teams.map((team) => 
    <TeamColumn
      key={team._id}
      id={team._id} 
      teamName={team.name} 
      project={team.project}
      teamMembers={team.teamMembers} />
  );

  return (
    <div className="team-columns">
      {teamColumns}
      <AddTeamColumn />
    </div>
  );
};

export default TeamColumns;