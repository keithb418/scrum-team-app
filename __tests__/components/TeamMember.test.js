import React from "react";
import { TeamMember } from "../../front-end/src/components/TeamMember";
import teamMembersData from '../fixtures/teamMembers';
import { shallow } from "enzyme";

let wrapper, handleDeleteTeamMember, button;
beforeEach(() => {
  handleDeleteTeamMember = jest.fn();
  wrapper = shallow(       
    <TeamMember 
      handleDeleteTeamMember={handleDeleteTeamMember}
      teamLead={teamMembersData[1].teamLead}
      id={teamMembersData[1].id}
      name={teamMembersData[1].name}
      role={teamMembersData[1].role}/>
  );
  button = wrapper.find('button');
});

test('should render the TeamMember component correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('the button should be draggable', () => {
  expect(button.props().onDragStart.length).toBe(1);
  expect(button.props().draggable).toBe("true");
});

test('should handle deleting team member', () => {
   wrapper.find('FontAwesome').at(0).simulate('click');
   expect(handleDeleteTeamMember).toHaveBeenCalledWith(teamMembersData[1].id);
});

