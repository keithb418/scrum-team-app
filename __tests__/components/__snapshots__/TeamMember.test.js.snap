// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the TeamMember component correctly 1`] = `
<button
  className="team-member btn btn-default "
  draggable="true"
  onDragStart={[Function]}
>
  <div
    className="team-member-header"
  >
    <p>
      Dennis
    </p>
    <FontAwesome
      className="delete-team-member-btn"
      name="trash"
      onClick={[Function]}
      tabIndex="1"
    />
  </div>
  <Link
    replace={false}
    to="/member/undefined"
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="user-img"
        componentClass="div"
        xs={4}
      >
        <FontAwesome
          name="user-circle"
        />
      </Col>
      <Col
        bsClass="col"
        className="details"
        componentClass="div"
        xs={8}
      >
        <p>
          Fullstack Developer
        </p>
      </Col>
    </Row>
  </Link>
</button>
`;
